org.idempiere.test/src/org/idempiere/test/base/QueryTest.java

```diff
@@ -333,4 +336,23 @@ public void testOnlySelection() throws Exception
 			assertEquals(expected, ids[i], "Element "+i+" not equals");
 		}
 	}
+
+	@Test
+	public void testVirtualColumnLoad() {
+		// create bogus record
+		MTest testPo = new MTest(Env.getCtx(), getClass().getName(), 1);
+		testPo.save();
+
+		BigDecimal expected = new BigDecimal(123.45d).setScale(2, RoundingMode.HALF_DOWN);
+
+		// test with virtual column lazy loading
+		Query query = new Query(Env.getCtx(), MTest.Table_Name, MTest.COLUMNNAME_Test_ID + "=?", getTrxName());
+		testPo = query.setParameters(testPo.get_ID()).first();
```

Comment:
Running this test on its own works, but when running all unit tests I am getting a Class Exception here:
java.lang.ClassCastException: class org.idempiere.test.model.MappedModelFactoryTest$MyTest2 cannot be cast to class org.compiere.model.MTest (org.idempiere.test.model.MappedModelFactoryTest$MyTest2 is in unnamed module of loader org.eclipse.osgi.internal.loader.EquinoxClassLoader @24949509; org.compiere.model.MTest is in unnamed module of loader org.eclipse.osgi.internal.loader.EquinoxClassLoader @67b28368)
	at org.idempiere.test.base.QueryTest.testVirtualColumnLoad(QueryTest.java:350)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)

Reply:
Good catch, it's fixed already. Thank you for testing.